crates/primitives/src/transaction/tx_type.rs

```diff
@@ -54,25 +64,38 @@ impl From<TxType> for U8 {
 }
 
 impl Compact for TxType {
-    fn to_compact<B>(self, _: &mut B) -> usize
+    fn to_compact<B>(self, buf: &mut B) -> usize
     where
         B: bytes::BufMut + AsMut<[u8]>,
     {
         match self {
             TxType::Legacy => 0,
             TxType::EIP2930 => 1,
             TxType::EIP1559 => 2,
-            TxType::EIP4844 => 3,
+            _ => {
```

Comment:
this match should be exhaustive

Reply:
These should already be exhaustive - The to/from compact impls for the `TxType` were altered to implement a positive lookahead if the 2 bits representing the tx type are both set. Would've been a DB breaking change if Cancun had already rolled around, but should be backwards compatible before then. If the first 2 bits are both set, this'll currently fall back to to a blob tx if the deposit tx type doesn't match on the next byte, which exhausts all tx type possibilities.

Reply:
i think matt's just saying to be explicit with `TxType::EIP4844 | TxType::DEPOSIT`?